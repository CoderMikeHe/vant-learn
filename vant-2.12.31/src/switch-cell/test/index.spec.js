import SwitchCell from '..';
import { mount } from '../../../test';

test('change event', () => {
  const onChange = jest.fn();
  const wrapper = mount(SwitchCell, {
    context: {
      on: {
        change: onChange,
      },
    },
  });

  wrapper.find('.van-switch').trigger('click');

  expect(onChange).toHaveBeenCalledWith(true);
});

test('border prop', () => {
  const wrapper = mount(SwitchCell, {
    propsData: {
      border: false,
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('cell-size prop', () => {
  const wrapper = mount(SwitchCell, {
    propsData: {
      cellSize: 'large',
    },
  });

  expect(wrapper).toMatchSnapshot();
});
