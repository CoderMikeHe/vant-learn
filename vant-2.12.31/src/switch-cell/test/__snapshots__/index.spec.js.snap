// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`border prop 1`] = `
<div class="van-cell van-cell--center van-cell--borderless van-switch-cell">
  <div class="van-cell__value van-cell__value--alone">
    <div role="switch" aria-checked="false" class="van-switch" style="font-size: 24px;">
      <div class="van-switch__node"></div>
    </div>
  </div>
</div>
`;

exports[`cell-size prop 1`] = `
<div class="van-cell van-cell--center van-cell--large van-switch-cell van-switch-cell--large">
  <div class="van-cell__value van-cell__value--alone">
    <div role="switch" aria-checked="false" class="van-switch" style="font-size: 24px;">
      <div class="van-switch__node"></div>
    </div>
  </div>
</div>
`;
